src/auth/oauth2client.ts

```diff
@@ -503,6 +504,30 @@ export class OAuth2Client extends AuthClient {
    */
   protected async refreshToken(refreshToken?: string|
                                null): Promise<GetTokenResponse> {
+    if (refreshToken) {
+      // If a request to refresh using the same token has started,
+      // return the same promise.
+      if (this.refreshTokenPromises.has(refreshToken)) {
+        return this.refreshTokenPromises.get(refreshToken)!;
+      }
+
+      const p = this.refreshTokenNoCache(refreshToken)
+                    .then(r => {
+                      this.refreshTokenPromises.delete(refreshToken);
+                      return r;
+                    })
+                    .catch(e => {
+                      this.refreshTokenPromises.delete(refreshToken);
```

Comment:
Is this catch path exercised by the tests?

Reply:
Nope.  Added one!